src/document/reducer.ts

```diff
@@ -226,69 +286,56 @@ export function baseReducer<T, A extends Action, L>(
     dispatch?: SignalDispatch,
     options: ReducerOptions = {},
 ) {
+    // if there's skip value
+    // 0:0 1:0 2:0 3:0 4:1(new action)
+    // garbagecollector(0:0 1:0 2:0 3:0 4:1)
+    // 0:0 1:0 2:0 4:1
+    // replayedDocument = replayOperation(0:0 1:0 2:0)
+    // newDocument = {...document, state: {...replayedDocument.state} }
+
+    // resutl = reducer(newDocument, 4:1)
+
     const { skip, ignoreSkipOperations = false, reuseHash = false } = options;
 
-    let _action = { ...action };
-    let skipValue = skip || 0;
-    let newDocument = document;
-    let clipboard = [...document.clipboard];
+    const _action = { ...action };
+    const skipValue = skip || 0;
+    let newDocument = { ...document };
+    // let clipboard = [...document.clipboard];
 
-    // if noop with skip value, validate if latest operation is the same as the current one
-    // if current one has a greather skip value, then replace the latest operation with the current one (drop latest op)
-    if (isNoopOperation(_action)) {
-        const { document: processedDocument, skip: calculatedSkip } =
-            noopOperation(newDocument, _action, skipValue);
+    const shouldProcessSkipOperation =
+        !ignoreSkipOperations &&
+        (skipValue > 0 || ('index' in _action && _action.skip > 0));
 
-        skipValue = calculatedSkip;
-        newDocument = processedDocument;
+    if (shouldProcessSkipOperation) {
```

Comment:
if there's an skip operation, replay document to calculate state before apply the action/operation

> NOTE: this line can be removed completely if we run garbageCollect against document operations before calling the reducer: https://github.com/powerhouse-inc/document-drive/blob/main/src/server/index.ts#L650